elastic/elastic-agent
# Add `os_family`, `os_platform` and `os_version` to host provider

## What does this PR do?
This commit adds `os_family`, `os_platform` and `os_version` to the host provider, enabling differentiating Linux distributions. This is required to support Debian 12 and other distributions that are moving away from traditional log files in favour of Journald.

## Why is it important?
It is required to support Debian 12 and other Linux distributions that are moving away from traditional log files in favour of Journald.

## Checklist


## Disruptive User Impact

None

## How to test this PR locally

1. Package the Elastic-Agent
2. Deploy it with any policy, even an empty policy with no monitoring will work
3. Collect the diagnostics
4. Look at `variables.yml` and ensure `os_family`, `os_platform` and `os_version` are populated under `host`. Here is an example from Debian 11:
```
      host:
        architecture: x86_64
        id: 9a7aa218396d47b1b17218a99918bae4
        ip:
            - 127.0.0.1/8
            - 10.0.2.15/24
            - ::1/128
            - fe80::a00:27ff:fe8d:c04d/64
        mac:
            - 08:00:27:8d:c0:4d
        name: bullseye
        os_family: debian
        os_platform: debian
        os_version: 11 (bullseye)
        platform: linux
```

## Related issues

- Relates https://github.com/elastic/integrations/issues/10797
- Is required by https://github.com/elastic/integrations/pull/11618

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
